layers/erspan2.go

```diff
@@ -0,0 +1,173 @@
+// https://tools.ietf.org/html/draft-foschiano-erspan-03
+type Header struct {
+	IsTruncated                         bool
+	Version, CoS, TrunkEncap            uint8
+	VLANIdentifier, SessionID, Reserved uint16
+	Index                               uint32
+}
+
+const (
+	// GREProtocolType - The protocol type value in the preceeding GRE header
+	GREProtocolType = 0x88BE
+	// VersionObsolete - The obsolete value for the version field
+	VersionObsolete = 0x0
+	// Version - The current value for the version field
+	Version = 0x1
+)
+
+// NewHeader is a convenience "constructor" which sets common default values
+func NewHeader() *Header {
+	return &Header{
+		Version: Version,
+	}
+}
+
+// DecodeFromBytes decodes the given bytes into this layer.
+func (h *Header) DecodeFromBytes(data []byte, df gopacket.DecodeFeedback) error {
+	h.Version = data[0] & 0xF0 >> 4
+	h.VLANIdentifier = binary.BigEndian.Uint16(data[:2]) & 0x0FFF
+	h.CoS = data[2] & 0xE0 >> 5
+	h.TrunkEncap = data[2] & 0x18 >> 3
+	h.IsTruncated = data[2]&0x4>>2 != 0
+	h.SessionID = binary.BigEndian.Uint16(data[2:4]) & 0x03FF
+	h.Reserved = binary.BigEndian.Uint16(data[4:6]) & 0xFFF0 >> 4
+	h.Index = binary.BigEndian.Uint32(data[4:8]) & 0x000FFFFF
+	return nil
+}
+
+// SerializeTo writes the serialized form of this layer into the
+// SerializationBuffer, implementing gopacket.SerializableLayer.
+// See the docs for gopacket.SerializableLayer for more info.
+func (h *Header) SerializeTo(b gopacket.SerializeBuffer, opts gopacket.SerializeOptions) error {
+	bytes, err := b.PrependBytes(8)
+	if err != nil {
+		return err
+	}
+
+	twoByteInt := uint16(h.Version&0xF)<<12 | h.VLANIdentifier&0x0FFF
+	binary.BigEndian.PutUint16(bytes, twoByteInt)
+
+	twoByteInt = uint16(h.CoS&0x7)<<13 | uint16(h.TrunkEncap&0x3)<<11 | h.SessionID&0x03FF
+	if h.IsTruncated {
+		twoByteInt |= 0x400
+	}
+	binary.BigEndian.PutUint16(bytes[2:], twoByteInt)
+
+	fourByteInt := uint32(h.Reserved&0x0FFF)<<20 | h.Index&0x000FFFFF
+	binary.BigEndian.PutUint32(bytes[4:], fourByteInt)
+	return nil
+}
+
+// Conn - A connection to a device to send packets within ERSPAN2 encapsulation
+type Conn struct {
+	ipConn *net.IPConn
+	opts   ConnOpts
+}
+
+// ConnOpts - ERSPAN2 connection options
+type ConnOpts struct {
+	MaxMTU uint16
+}
+
+// Dial - Connect to a device given an IP address string.
+func Dial(addr string, opts *ConnOpts) (*Conn, error) {
+	ipAddr, err := net.ResolveIPAddr("ip", addr)
+	if err != nil {
+		return nil, err
+	}
+
+	conn, err := net.DialIP("ip:gre", nil, ipAddr)
+	if err != nil {
+		return nil, err
+	}
+
+	if opts == nil || opts.MaxMTU == 0 {
+		opts = &ConnOpts{MaxMTU: 1500}
+	}
+	return &Conn{
+		ipConn: conn,
+		opts:   *opts,
+	}, nil
+}
+
+// IPConn provides access to the IPConn associated with this ERSPANConn
+func (e *Conn) IPConn() *net.IPConn {
+	return e.ipConn
+}
+
+// Options provides access to a copy of the connection options
+func (e *Conn) Options() ConnOpts {
+	return e.opts
+}
+
+const (
```

Comment:
If needed, please use the const already defined in Gopacket other layers (ie IP / Ethernet)